typescript/sdk/src/hook/HyperlaneHookDeployer.ts

```diff
@@ -124,4 +131,38 @@ export class HyperlaneHookDeployer extends HyperlaneDeployer<
     );
     return hooks;
   }
+
+  async deployOpStack(
+    chain: ChainName,
+    config: OpStackHookConfig,
+    coreAddresses = this.core[chain],
+  ): Promise<HyperlaneContracts<HookFactories>[HookType.OP_STACK]> {
+    const mailbox = coreAddresses.mailbox;
+    if (!mailbox) {
+      throw new Error(`Mailbox address is required for ${config.type}`);
+    }
+    // fetch l2 messenger address from l1 messenger
+    const l1Messenger = IL1CrossDomainMessenger__factory.connect(
+      config.nativeBridge,
+      this.multiProvider.getSignerOrProvider(chain),
+    );
+    const l2Messenger: Address = await l1Messenger.OTHER_MESSENGER();
+    // deploy opstack ism
+    const ismConfig: OpStackIsmConfig = {
+      type: ModuleType.OP_STACK,
+      nativeBridge: l2Messenger,
+    };
```

Comment:
doesn't the ISM need the hook address?

Reply:
ismConfig doesn't need to have hook address bc setAuthorizedHook is after the hook deployment